### PR TITLE
use npm for jqueryui

### LINKED ISSUE #5580: Fetch jquery-ui from NPM
part of #5458

Look out for javascript errors throughout the (non-mobile) site.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #5580 
part of #5458

This upgrades jquery-ui as well as switching to npm (our old version wasn't available).   There were some packaging changes so files have moved, and the old core.js is now separate files (easy when you use a modern module bundler... annoying with our setup).  Also drops jquery-ui css bundle because it would have been a lot of extra files that were only used in /admin - I can't see it made any difference anyway.

### CODE REVIEW

File: src/olympia/lib/settings_base.py [hunk 1]
```diff
@@ -683,20 +681,34 @@ def JINJA_CONFIG():
...
-            'js/lib/jquery-ui/datepicker.js',
-            'js/lib/jquery-ui/sortable.js',
+            'js/node_lib/ui/version.js',
+            'js/node_lib/ui/data.js',
+            'js/node_lib/ui/disable-selection.js',
+            'js/node_lib/ui/ie.js',
+            'js/node_lib/ui/keycode.js',
+            'js/node_lib/ui/escape-selector.js',
+            'js/node_lib/ui/labels.js',
+            'js/node_lib/ui/jquery-1-7.js',
```
Comment (muffinresearch) [inline src/olympia/lib/settings_base.py]: Is this jquery itself?

Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: I did jquery in https://github.com/mozilla/addons-server/pull/5530

**Comment by muffinresearch** [comment]
Might be nice to check the bundle sizes before and after because that could be an indicator of this including more than it needs to in splitting it out.

**Comment by eviljeff** [comment]
Do you mean just look on -dev for the rough size?  The contents aren't directly comparable because it's an upgrade where they changed a lot of files.

File: src/olympia/lib/settings_base.py (hunk 1 above)
Comment (muffinresearch) [inline src/olympia/lib/settings_base.py]: Is this something only related to ui then? Just wanted to make sure this isn't an actual jquery lib because presumably we're already bringing that in elsewhere?

**Comment by muffinresearch** [comment]
> Do you mean just look on -dev for the rough size? The contents aren't directly comparable because it's an upgrade where they changed a lot of files.

Yeah I mean compare the built file size to make sure it's not massively different.

File: src/olympia/lib/settings_base.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: Oh, that file specifically.  It's a helper file jquery-ui uses to hook into jquery itself - oddly named imo.

Comment (muffinresearch) [inline src/olympia/lib/settings_base.py]: Ah ok - that's fine then 👍

File: Makefile [hunk 2]
```diff
@@ -155,7 +157,7 @@ copy_node_js:
 ifeq ($(IN_DOCKER),)
 	$(warning Command is designed to be run in the container)
 endif
-	for dest in $(NODE_LIBS_JS) ; do cp $(NODE_MODULES)$$dest $(STATIC_JS) ; done
+	for dest in $(NODE_LIBS_JS) ; do cp -r $(NODE_MODULES)$$dest $(STATIC_JS) ; done
```
Comment (muffinresearch) [inline Makefile]: Would it be possible to make this explict rather than copying entire directories? I'd have concerns that this way we don't really know what we're putting on the CDN.

Comment (muffinresearch) [inline Makefile]: I suppose an alternative would be to use the lists in settings to do the copying but that's a bit of a yak-shaving excercise.

File: src/olympia/lib/settings_base.py [hunk 3]
```diff
@@ -763,14 +775,28 @@ def JINJA_CONFIG():
...
+            'js/node_lib/ui/focusable.js',
+            'js/node_lib/ui/tabbable.js',
+            'js/node_lib/ui/unique-id.js',
+            'js/node_lib/ui/position.js',
+            'js/node_lib/ui/widget.js',
+            'js/node_lib/ui/widgets/mouse.js',
+            'js/node_lib/ui/widgets/menu.js',
+            'js/node_lib/ui/widgets/autocomplete.js',
+            'js/node_lib/ui/widgets/datepicker.js',
+            'js/node_lib/ui/widgets/sortable.js',
```
Comment (EnTeQuAk) [inline src/olympia/lib/settings_base.py]: We're packing in now more into our bundle than we originally did which should increase it's size. Do we really need all of them?

Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: More files, but the same content.  core.js was [split](https://github.com/jquery/jquery-ui/blob/1.12.1/ui/core.js) into 15 different files.  I did a quick search for each of the files, and they all _appeared_ to be used, or at least referenced in our js - working out whether all of _our_ js is still used is another issue.

Comment (EnTeQuAk) [inline src/olympia/lib/settings_base.py]: Ah, that makes sense. Works for me :)

File: Makefile (hunk 2 above)
Comment (eviljeff) [inline Makefile]: > Would it be possible to make this explict rather than copying entire directories? I'd have concerns that this way we don't really know what we're putting on the CDN.

It's possible, it'd just mean more that needs to be updated in the makefile each time.  I'll make the change.

> I suppose an alternative would be to use the lists in settings to do the copying but that's a bit of a yak-shaving excercise.

That would be a little more difficult because the lists reference the destination of the files (`node_lib`) rather than the source.

I thought about trying to reference them directly from the node folders in the lists (a relative reference to files), rather than copying at all, but a) seemed fragile on different platforms where the `node_modules` folder might be somewhere different and b) I wasn't sure if having the source in one place for reference was desirable rather than having to go wading through `node_modules`.

**Comment by eviljeff** [comment]
> Yeah I mean compare the built file size to make sure it's not massively different.

I compared `common-min.js` and `impala-min.js` with prod and the new ones are each 4kb bigger (213kb vs 209kb and 246kb vs 242kb).  I can live with a 2% increase for the new version.

File: Makefile (hunk 2 above)
Comment (eviljeff) [inline Makefile]: https://github.com/mozilla/addons-server/pull/5581/commits/b38912c42a998c1af478c0fd54d4fc914f0c54fb
(I rebased it into a single commit due to merge conflict issues)
